### PR TITLE
'File partition' option and 'document' directory specification

### PR DESCRIPTION
**Pull request by Whattabatt**
Allows for partitioning work at the file level - checking the hash of the de-globbed file names instead of spans. Also, creates an optional 'document dir' parameter for dealing with source data that doesn't have a 'document' directory to replace with 'attributes' (previously the attribute path generator would try replacing 'documents', not find any, and use the source directory!)

### CODE REVIEW

File: src/deduper.rs [hunk 1]
```diff
@@ -123,7 +137,15 @@ fn write_attributes(
...
-        docs_location.replace("/documents/", &attr_prefix)
+        docs_location.replace(
+            &format!(
+                "/{}/",
+                dedupe_config
+                    .document_dir
+                    .unwrap_or(String::from("documents"))
+            ),
+            &attr_prefix,
+        )
```
Comment (undfined) [inline src/deduper.rs:148]: Should we add a test for this?

File: pyproject.toml [hunk 2]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "dolma"
-version = "1.0.14.post1"
+version = "1.0.14.dev6"
```
Comment (undfined) [inline pyproject.toml:3]: Let's update the version and drop devN

**Review by undfined — APPROVED** [review]
Looks good, likely worth a test given the destructive nature of the bug you found.

File: pyproject.toml [hunk 3]
```diff
@@ -1,6 +1,6 @@
 [project]
 name = "dolma"
-version = "1.0.14.dev6"
+version = "1.0.5"
```
[the remark this answers is not in the document]
Comment (Whattabatt) [reply pyproject.toml:3]: No actually, when I started making dev branches I misread the version. The most recent not dev is 1.0.4

**Pull request merged**